sdk/servicebus/service-bus/src/core/linkEntity.ts

```diff
@@ -217,12 +217,19 @@ export abstract class LinkEntity<LinkT extends Receiver | AwaitableSender | Requ
     this._logger.verbose(
       `${this._logPrefix} Attempting to acquire lock token ${this._openLock} for initializing link`
     );
-    return defaultLock.acquire(this._openLock, () => {
-      this._logger.verbose(
-        `${this._logPrefix} Lock ${this._openLock} acquired for initializing link`
-      );
-      return this._initLinkImpl(options, abortSignal);
-    });
+    return defaultCancellableLock.acquire(
```

Comment:
I want to think about this one more. 

It opens up an execution path that did not exist before. It's now possible for _two_ _initLinkImpl's to be executing at the same time if the outer lock aborts and the inner operation is still running. 

This complicates the code for _initLinkImpl a bit because it now has to be careful when unwinding state that it's not conflicting with another call of itself (which wasn't a concern before).

Do we need this change?

Reply:
The abortSignal can only cause the acquire call to abort if the inner function hasn't been invoked yet. As soon as that inner function is invoked, both the timeout and abortSignal on `acquire` are cancelled/removed.

Does that clear up your concerns?

Reply:
It does.

Looking at the doc comment the behavior you described is pretty clear for 'task', but less clear for 'abort'.